akka-typed/src/main/scala/akka/typed/ScalaProcess.scala

```diff
@@ -0,0 +1,538 @@
+    extends Operation[Nothing, a.Cancellable] with AbstractWatchRef {
+    type Msg = T
+    override def equals(other: Any) = super.equals(other)
+    override def hashCode() = super.hashCode()
+  }
+  private[typed] case class Replay[T](key: StateKey[T]) extends Operation[Nothing, T]
+  private[typed] case class Snapshot[T](key: StateKey[T]) extends Operation[Nothing, T]
+  private[typed] case class State[S, T <: StateKey[S], E](key: T, afterUpdates: Boolean, transform: S ⇒ (Seq[T#Event], E)) extends Operation[Nothing, E]
+  private[typed] case class StateR[S, T <: StateKey[S]](key: T, afterUpdates: Boolean, transform: S ⇒ Seq[T#Event]) extends Operation[Nothing, S]
+  private[typed] case class Forget[T](key: StateKey[T]) extends Operation[Nothing, akka.Done]
+
+  // FIXME figure out cleanup of external resources after a failure
+
+  /*
+   * The core operations: keep these minimal!
+   */
+
+  /**
+   * Obtain a reference to the ActorSystem in which this process is running.
+   */
+  def opSystem(implicit opDSL: OpDSL): Operation[opDSL.Self, ActorSystem[Nothing]] =
+    System
+
+  /**
+   * Read a message from this process’ input channel.
+   */
+  def opRead(implicit opDSL: OpDSL): Operation[opDSL.Self, opDSL.Self] =
+    Read
+
+  /**
+   * Obtain this process’ [[ActorRef]], not to be confused with the ActorRef of the Actor this process is running in.
+   */
+  def opProcessSelf(implicit opDSL: OpDSL): Operation[opDSL.Self, ActorRef[opDSL.Self]] =
+    ProcessSelf
+
+  /**
+   * Obtain the [[ActorRef]] of the Actor this process is running in.
+   */
+  def opActorSelf(implicit opDSL: OpDSL): Operation[opDSL.Self, ActorRef[ActorCmd[Nothing]]] =
+    ActorSelf
+
+  /**
+   * Lift a plain value into a process that returns that value.
+   */
+  def opUnit[U](value: U)(implicit opDSL: OpDSL): Operation[opDSL.Self, U] =
+    Return(value)
+
+  /**
+   * Execute the given process within the current Actor, await and return that process’ result.
+   */
+  def opCall[Self, Out](process: Process[Self, Out])(implicit opDSL: OpDSL): Operation[opDSL.Self, Out] =
+    Call(process)
+
+  /**
+   * Create and execute a process with a self reference of the given type,
+   * await and return that process’ result. This is equivalent to creating
+   * a process with [[OpDSL]] and using `call` to execute it.
+   */
+  def opNextStep[T] =
+    OpDSL.nextStep.asInstanceOf[OpDSL.NextStep[T]]
+
+  /**
+   * Execute the given process within the current Actor, concurrently with the
+   * current process. The value computed by the forked process cannot be
+   * observed, instead you would have the forked process send a message to the
+   * current process to communicate results. The returned [[SubActor]] reference
+   * can be used to send messages to the forked process or to cancel it.
+   */
+  def opFork[Self](process: Process[Self, Any])(implicit opDSL: OpDSL): Operation[opDSL.Self, SubActor[Self]] =
```

Comment:
Ah, excellent. Was about to comment that "oh but we'll need a new verb because child actors..." but fork already is that new verb. Very nice actually – fork fits the SubActor well as does spawn for children. Thumbs up here.